source/extensions/listener_managers/listener_manager/filter_chain_manager_impl.h

```diff
@@ -100,6 +111,14 @@ class PerFilterChainFactoryContextImpl : public Configuration::FilterChainFactor
   Stats::ScopeSharedPtr filter_chain_scope_;
   Init::Manager& init_manager_;
   std::atomic<bool> is_draining_{false};
+  // TODO(ramaraochavali): Move DownstreamFilterConfigProviderManager usages to use this instead of
+  // current singleton.
+  std::shared_ptr<Filter::FilterConfigProviderManager<Filter::NamedHttpFilterFactoryCb,
```

Comment:
I think this might be a problem - there is no re-use of the dynamic providers between ECDS in listeners and ECDS in composite filter, so if they use the same name, there'll be two instances of the filter. Can we include the downstream filter config provider unification into this PR?

Reply:
I can do that or separate it out in a different PR and we can merge that first. WDYT? @ohadvano Do you have any concerns with that?

Reply:
I think it makes sense to first do a PR that changes the behavior to use the manager stored in the factory context